src/app/theme/services/theme.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from "@angular/core";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ThemeService {
+
+  public static default = 'light';
+
+  public get current(): string {
+    return localStorage.getItem('theme') ?? ThemeService.default;
+  }
+
+  public set current(value: string) {
+    localStorage.setItem('theme', value);
+    this.style.href = `${value}.css`;
+  }
+
+  private readonly style: HTMLLinkElement;
+
+  constructor() {
+    this.style = document.createElement('link');
+    this.style.rel = 'stylesheet';
+    document.head.appendChild(this.style);
```

Comment:
Not really a fan of dynamically loading the CSS when needed.
Could you rework this a bit and have all CSS styles served directly to the user?

You can use a CSS selector to detect if the browser is using darkmode:
```css
@media (prefers-color-scheme: dark) {
    /* Rules here */
}
```

Or, you could set a class on the body to indicate the user has switched to dark mode.
Eg: `<body class="dark-mode">`